src/catalog/catalog.ts

```diff
@@ -107,6 +118,14 @@ export class Catalog implements ICatalog {
         return this.mapSpecific.has(key);
     }
 
+    getKeyForSku(sku: SKU): Key {
+        const key = this.skuToKey.get(sku);
+        if (!key) {
```

Comment:
do you want to use this.skuToKey.has(sku)  method to find out if the entry exists before retrieving it.

Reply:
Since he's returning the key I don't think that's necessary.